proc_scripts/CPMG_phasing.py

```diff
@@ -11,7 +10,7 @@ def center_CPMG_echo(s, axis='t2',fl=None):
     s /= zeroth_order_ph(s[axis:0])
     time_bound = min(abs(s.getaxis(axis)[r_[0,-1]]))
     s = s[axis:(-time_bound,time_bound)]
-    assert math.isclose(s.getaxis(axis)[1],-s.getaxis(axis)[-1],abs_tol=0.05),"echo is not symmetric! you are using the wrong code!!"
+    assert isclose(s.getaxis(axis)[0],-s.getaxis(axis)[-1]),"echo is not symmetric! you are using the wrong code!!"
     return s
```

Comment:
I'm fixing a ROLLBACK ☹☹☹☹☹☹☹!

Reply:
[doc](https://numpy.org/doc/stable/reference/generated/numpy.isclose.html)

Reply:
the tolerance needs to be clarified. without it It says the echo is not symmetric even though I tested what the value of s.getaxis(axis)[0] and s.getaxis(axis)[-1] and they are the same to 0.00001 but the default of is close is 1e-9 so we need this adjusted to be able to run.

Reply:
I don't know why the documentation I saw was different than yours but abs_tol = atol in your case

Reply:
You changed the first index to 1, when it should be zero, and numpy (which I ref. above already has a function, and you can adjust the tolerance).

Also, your absolute tolerance number is way too big -- the default is 1e-8, which is more than big enough (for typical dwell times, that's going to be at least a full percent of the dwell time, if not more)

Reply:
Okay so I went in and placed print statements for printing the indices we are using for the assert statement:
```
this is what s.getaxis(axis)[0] is:
-0.015250930842946958
this is what s.getaxis(axis)[-1] is:
0.015500946102667397
```
This is a dataset that Alec had run with the CPMG spincore as you can see these are not close by e-8 or e-9. Hence why I believe the atol is needed. I will try with a separate dataset and test it out though to double check

Reply:
okay so I tried with another CPMG file... really weird but it gives the SAME EXACT values. no difference...

Reply:
If you're saying they have the same time axis, that's possible.

Reply:
But to the bigger point, it seems like it should be throwing this error. Look at the difference between the first and second elements of the array, to see what the spacing (dwell time) is. I bet the discrepancy here is about 1 dwell time.

Reply:
So when I take the first and second elements of the array ([0] and [1]) I get:
```
this is what s.getaxis(axis)[0] is:
-0.015250930842946958
this is what s.getaxis(axis)[1] is:
-0.015000915583226516
```
So it seems this data has a relatively large dwell time (~0.00025) I am going to set the tolerance to this number for now

Reply:
So we definitely should be getting an error because the time axis is not symmetric. The one side is a full dwell time further from zero than the other side.

Reply:
This actually shouldn't be happening. Are you running the version of pyspecdata that has the upgrades to parse slices?

Reply:
The reason that shouldn't be happening is that up above in the code you hold it to make the axis in register with zero so that there is a point that is exactly equal to zero. Then you told it to take a symmetric slice. Given those requirements, the first point in your axis coordinates should be exactly equal to the negative of the last point. I asked the question about the library version because older versions did not slice inclusively. so please check the library version check that you still have the register axis line, and then let me know which script I need to run to reproduce this error.

Reply:
OK!  So, first, I edited the code to put any type of failure information into the assertion error

Reply:
Then, I used this to debug parse_slices in pyspecdata (which I had changed recently to be less ambiguous and to use more efficient underlying numpy).

Reply:
So, if you pull the most recent version of pyspecdata master, this should work now.

Reply:
for dataset '200221_CPMG_TEMPOLgel_2p9_1' I get the error 
```AssertionError: echo is not symmetric! you are using the wrong code!! (first point is -0.0152509, last point 0.0155009,
dwell time 0.000250015, and time_bound=0.0152509```
for dataset '200304_CPMG_2p6_1','test_equip' I get the error
```AssertionError: echo is not symmetric! you are using the wrong code!! (first point is -0.0147509, last point 0.0150009,
dwell time 0.000250015, and time_bound=0.0147509```
for dataset '200305_CPMG_3p5_2'
```AssertionError: echo is not symmetric! you are using the wrong code!! (first point is -0.015751, last point 0.016001, dw
ell time 0.000250015, and time_bound=0.015751```
for dataset ''200305_CPMG_3p6_2'
```AssertionError: echo is not symmetric! you are using the wrong code!! (first point is -0.0155009, last point 0.015751, d
well time 0.000250015, and time_bound=0.0155009```
for dataset '200305_CPMG_3p7_2'
```AssertionError: echo is not symmetric! you are using the wrong code!! (first point is -0.0152509, last point 0.0155009,
dwell time 0.000250015, and time_bound=0.0152509```
for dataset '200305_CPMG_3p7_3'
```AssertionError: echo is not symmetric! you are using the wrong code!! (first point is -0.0155009, last point 0.015751, d
well time 0.000250015, and time_bound=0.015751```
for dataset '200305_CPMG_3p8_2'
```AssertionError: echo is not symmetric! you are using the wrong code!! (first point is -0.0155009, last point 0.015751, d
well time 0.000250015, and time_bound=0.015751```
for dataset '200305_CPMG_3p9_2'
```AssertionError: echo is not symmetric! you are using the wrong code!! (first point is -0.0150009, last point 0.0152509,
dwell time 0.000250015, and time_bound=0.0150009```
for dataset '200305_CPMG_4p0_1'
```AssertionError: echo is not symmetric! you are using the wrong code!! (first point is -0.0155009, last point 0.015751, d
well time 0.000250015, and time_bound=0.015751```
my point is none of these can run with the tolerance set to 1e-8 or 1e-9. This is all of the datasets for the spincore CPMG of late that I know of

Reply:
I just checked 200305_CPMG_3p7_2 on the most recent version of your code.  You didn't pull the most recent version of pyspecdata.

Reply:
If you look at the numbers above, they are all off by 1 dwell time.  The tolerance is not the issue.